File: HaberUygulamaJava/app/src/main/java/com/example/haberuygulamajava/model/YorumModel.java
package com.example.haberuygulamajava.model;

import java.io.Serializable;

public class YorumModel implements Serializable {
    private int id;
    private int haber_id;
    private String takma_ad;
    private String yorum_metni;
    private int onayli;
    private String yorum_tarihi;

    public YorumModel() {}

    public YorumModel(int id, int haber_id, String takma_ad, String yorum_metni, int onayli, String yorum_tarihi) {
        this.id = id;
        this.haber_id = haber_id;
        this.takma_ad = takma_ad;
        this.yorum_metni = yorum_metni;
        this.onayli = onayli;
        this.yorum_tarihi = yorum_tarihi;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getHaber_id() {
        return haber_id;
    }

    public void setHaber_id(int haber_id) {
        this.haber_id = haber_id;
    }

    public String getTakma_ad() {
        return takma_ad;
    }

    public void setTakma_ad(String takma_ad) {
        this.takma_ad = takma_ad;
    }

    public String getYorum_metni() {
        return yorum_metni;
    }

    public void setYorum_metni(String yorum_metni) {
        this.yorum_metni = yorum_metni;
    }

    public int getOnayli() {
        return onayli;
    }

    public void setOnayli(int onayli) {
        this.onayli = onayli;
    }

    public String getYorum_tarihi() {
        return yorum_tarihi;
    }

    public void setYorum_tarihi(String yorum_tarihi) {
        this.yorum_tarihi = yorum_tarihi;
    }
}

File: flutter_haber_uygulama/lib/model/yorum_model.dart
// yorum_model.dart
class YorumModel {
  final int id;
  final int haber_id;
  final String takma_ad;
  final String yorum_metni;
  final int onayli;
  final String yorum_tarihi;

  const YorumModel({
    required this.id,
    required this.haber_id,
    required this.takma_ad,
    required this.yorum_metni,
    required this.onayli,
    required this.yorum_tarihi,
  });

  factory YorumModel.fromJson(Map<String, dynamic> json) {
    return YorumModel(
      id: (json['id'] ?? 0) as int,
      haber_id: (json['haber_id'] ?? 0) as int,
      takma_ad: (json['takma_ad'] ?? '') as String,
      yorum_metni: (json['yorum_metni'] ?? '') as String,
      onayli: (json['onayli'] ?? 0) as int,
      yorum_tarihi: (json['yorum_tarihi'] ?? '') as String,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'haber_id': haber_id,
    'takma_ad': takma_ad,
    'yorum_metni': yorum_metni,
    'onayli': onayli,
    'yorum_tarihi': yorum_tarihi,
  };
}
